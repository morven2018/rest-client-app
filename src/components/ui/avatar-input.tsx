import { useRef, useState } from 'react';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';

interface AvatarInputProps {
  onAvatarChange?: (file: File | null) => void;
  placeholder?: string;
}

export function AvatarInput({
  onAvatarChange,
  placeholder = 'Choose an avatar...',
}: Readonly<AvatarInputProps>) {
  const [fileName, setFileName] = useState<string>('');
  const fileInputRef = useRef<HTMLInputElement>(null);

  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const file = event.target.files?.[0];
    if (file?.type.startsWith('image/')) {
      setFileName(file.name);
      onAvatarChange?.(file);
    } else {
      setFileName('');
      onAvatarChange?.(null);
    }
  };

  const handleButtonClick = () => {
    fileInputRef.current?.click();
  };

  return (
    <div className="flex w-full max-w-sm items-center gap-2">
      <Input
        type="text"
        placeholder={placeholder}
        value={fileName}
        readOnly
        className="flex-1"
      />
      <Input
        ref={fileInputRef}
        type="file"
        accept="image/*"
        onChange={handleFileChange}
        className="hidden"
      />

      <Button type="button" variant="outline" onClick={handleButtonClick}>
        Browse
      </Button>
    </div>
  );
}
