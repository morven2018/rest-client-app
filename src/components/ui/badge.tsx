import * as React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';
import { Loader } from 'lucide-react';
import { cn } from '@/lib/utils';

const badgeVariants = cva(
  'inline-flex items-center justify-center rounded-lg border-transparent text-white border px-4 py-1 text-base font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-4 gap-2.5 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden',
  {
    variants: {
      variant: {
        default: 'bg-gray-400',
        ok: 'bg-teal-600',
        error: 'bg-red-600',
        info: 'bg-indigo-600',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  }
);

function Badge({
  className,
  variant,
  asChild = false,
  ...props
}: React.ComponentProps<'span'> &
  VariantProps<typeof badgeVariants> & { asChild?: boolean }) {
  const Comp = asChild ? Slot : 'span';

  return (
    <Comp
      data-slot="badge"
      className={cn(badgeVariants({ variant }), className)}
      {...props}
    >
      {variant === 'info' && <Loader className="size-4" />}
      {props.children}
    </Comp>
  );
}

export { Badge, badgeVariants };
