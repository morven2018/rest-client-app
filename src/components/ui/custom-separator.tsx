import * as React from 'react';
import { cn } from '@/lib/utils';

interface SeparatorProps {
  className?: string;
  children?: React.ReactNode;
}

export function CustomSeparator({
  className,
  children,
}: Readonly<SeparatorProps>) {
  return (
    <div
      className={cn('relative flex items-center py-4', className)}
      data-testid="separator"
    >
      <div className="flex-grow border-t border-border"></div>
      {children && (
        <span className="mx-3 flex-shrink text-sm text-muted-foreground">
          {children}
        </span>
      )}
      <div className="flex-grow border-t border-border"></div>
    </div>
  );
}
