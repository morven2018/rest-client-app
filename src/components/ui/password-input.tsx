'use client';
import * as React from 'react';
import { Eye, EyeOff } from 'lucide-react';
import { Button } from './button';
import { Input } from './input';
import { cn } from '@/lib/utils';

const PasswordInput = React.forwardRef<
  HTMLInputElement,
  React.InputHTMLAttributes<HTMLInputElement>
>(({ className, ...props }, ref) => {
  const [showPassword, setShowPassword] = React.useState(false);

  const handleTogglePasswordVisibility = () => {
    setShowPassword((prev) => !prev);
  };

  return (
    <div className="relative">
      <Input
        ref={ref}
        type={showPassword ? 'text' : 'password'}
        className={cn('pr-10', className)}
        {...props}
      />
      <Button
        type="button"
        variant="ghost"
        size="icon"
        className="absolute right-0 top-0 h-full px-3 py-2 hover:bg-transparent"
        onClick={handleTogglePasswordVisibility}
      >
        {showPassword ? (
          <EyeOff className="size-4 text-muted-foreground" />
        ) : (
          <Eye className="size-4 text-muted-foreground" />
        )}
        <span className="sr-only">
          {showPassword ? 'Hide password' : 'Show password'}
        </span>
      </Button>
    </div>
  );
});

PasswordInput.displayName = 'PasswordInput';

export { PasswordInput };
