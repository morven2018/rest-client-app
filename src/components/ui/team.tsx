import React from 'react';
import yuliaImage from '../../../public/team-img/junior.jpg';
import igorImage from '../../../public/team-img/Igor.jpg';
import alenaImage from '../../../public/team-img/Alena.jpg';
import Image, { StaticImageData } from 'next/image';

interface TeamMember {
  name: string;
  image: StaticImageData;
  description: string;
  bio: string[];
  contributions: string[];
  gitHubUrl: string;
}

const TeamMembers = () => {
  const teamMembers: TeamMember[] = [
    {
      name: 'Alena Pudina',
      image: alenaImage,
      description: 'Team Lead | UI/UX Design | Frontend Development',
      bio: [
        'Math & Computer Science Genius.',
        "Skilled Programmer with a Master's Degree in Applied Mathematics and Computer Science.",
        'Excelled in Physics and Chemistry Olympiads during her School Years.',
        'Experience as a Project Manager.',
      ],
      contributions: [
        'Led UI/UX design implementation in Figma.',
        'Established Project folder structure and Layout Architecture.',
        'Implemented Header component and Registration Module.',
        'Configured dynamic Routing for auto-generated pages.',
        'Developed User Profile page with interactive functionality.',
        'Built shopping Cart Page with order processing functionality.',
        'Created robust Validation System for form inputs and user data.',
        'Fixed SonarQube code quality issues and integrated mentors suggestions to optimize performance.',
        'Conducted code reviews to foster successful collaboration among developers, significantly enhancing code quality by identifying and resolving issues.',
      ],
      gitHubUrl: 'https://github.com/morven2018',
    },
    {
      name: 'Igor Batura',
      image: igorImage,
      description: 'Senior Fullstack Developer',
      bio: [
        'Natural-Born Leader with a Vision for Technical Innovation.',
        'Extremely Talented & Skilled Fullstack Developer with a Strong Work Ethic.',
        'Resilient, Disciplined & Goal-oriented.',
        'Collaborative Team Player Who Elevates Those Around Him and Amplifies Collective Success.',
      ],
      contributions: [
        'Partnered with the Team Lead on architectural decisions and task delegation.',
        'Architected and configured all project configuration files.',
        'Designed and implemented Backend infrastructure with secure API endpoints and database connectivity.',
        'Developed product Catalog Page with dynamic filtering, sorting, and pagination functionality.',
        'Engineered secure login Authentication system with form validation and error handling.',
        'Fixed SonarQube code quality issues and integrated mentors suggestions to optimize performance.',
        'Implemented a Router for page navigation and redirecting authorized/unauthorized users based on manual browser URL changes.',
        'Reviewed Junior Developer code, providing constructive feedback and mentorship.',
      ],
      gitHubUrl: 'https://github.com/Ihar-Batura',
    },
    {
      name: 'Yulia Podgurskaia',
      image: yuliaImage,
      description:
        'Project Manager | English Translation | Automated Testing | Frontend Development',
      bio: [
        'Trilingual Frontend Developer: English, Chinese, Russian.',
        'Excelled in Math, Chemistry, Russian, English Olympiads and Music Competitions.',
        'Received a Medal for Academic Excellence. Graduated with Two Diplomas with Honors.',
        'Pianist, Dancer & Athlete.',
      ],
      contributions: [
        'Collaborated on UI/UX design system implementation.',
        'Built several sections, pages and components (Promo Code, About Page, Besteseller, Popular, Strengths, Card Component, Footer).',
        'Orchestrated sprint planning via Trello task tracking and executed production deployments.',
        'Configured Commerce Tools platform, optimizing product catalog management.',
        'Established comprehensive testing suite with 30%+ coverage using Jest, Vitest, and React Testing Library.',
        'Conducted manual testing on multiple browsers and devices for reliability.',
        'Refactored Codebase, fixed SonarQube issues, integrated mentors code review suggestions.',
        'Conducted code reviews to promote effective team collaboration and improve code quality.',
      ],
      gitHubUrl: 'https://github.com/yuliafire',
    },
  ];

  const handleGitHubClick = (url: string, event: React.MouseEvent) => {
    event.stopPropagation();
    window.open(url, '_blank', 'noopener, noreferrer');
  };

  return (
    <div className="w-full bg-gray-50 dark:bg-gray-900 py-12 px-4 sm:px-6 lg:px-8">
      <h2 className="text-3xl md:text-4xl font-bold text-gray-900 dark:text-gray-100 text-center mb-8">
        Our Team
      </h2>
      <div className="w-full mx-auto space-y-4">
        {teamMembers.map((member, index) => (
          <div
            key={index}
            className="max-w-[1440px] mx-auto flex flex-col md:flex-row bg-gradient-to-r from-[#1C2024] to-[#8B8D98] rounded-lg shadow-lg overflow-hidden hover:shadow-xl transition-shadow duration-300"
          >
            <div className="md:w-1/3 p-6 flex flex-col items-center justify-center">
              <div className="relative w-48 h-48 mb-4">
                <Image
                  src={member.image}
                  alt={member.name}
                  fill
                  className="object-cover border-4 border-gray-200 dark:border-gray-700 rounded-[20px]"
                />
              </div>
              <div className="text-center">
                <div className="flex items-center justify-center space-x-2 mb-2">
                  <a
                    href={member.gitHubUrl}
                    onClick={(e) => handleGitHubClick(member.gitHubUrl, e)}
                    target="_blank"
                    rel="noopener noreferrer"
                    aria-label={`Visit ${member.name}'s GitHub profile`}
                    className="text-xl font-semibold text-white-600 dark:text-white-400 hover:underline"
                  >
                    {member.name}
                  </a>
                  <a
                    href={member.gitHubUrl}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="text-purple-600 dark:text-purple-400 hover:text-purple-700 dark:hover:text-purple-300"
                  >
                    <svg
                      className="w-5 h-5"
                      fill="currentColor"
                      viewBox="0 0 16 16"
                    >
                      <path d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.03 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z" />
                    </svg>
                  </a>
                </div>
                <p className="text-gray-600 dark:text-gray-400 text-sm mb-4">
                  {member.description}
                </p>
              </div>
            </div>

            <div className="md:w-2/3 p-6 border-t md:border-t-0 md:border-l border-gray-200 dark:border-gray-700">
              <div className="mb-6">
                <h3 className="text-lg font-semibold text-gray-900 dark:text-gray-100 mb-3">
                  About
                </h3>
                <div className="text-gray-600 dark:text-gray-300 text-sm space-y-2 text-left">
                  {member.bio.map((line, i) => (
                    <p key={i}>{line}</p>
                  ))}
                </div>
              </div>

              <div>
                <h3 className="text-lg font-semibold text-gray-900 dark:text-gray-100 mb-3">
                  Contributions
                </h3>
                <ul className="text-gray-600 dark:text-gray-300 text-sm space-y-2">
                  {member.contributions.map((contribution, i) => (
                    <li key={i} className="flex items-start text-left">
                      <span className="text-green-500 mr-2 mt-1">âœ“</span>
                      <span>{contribution}</span>
                    </li>
                  ))}
                </ul>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default TeamMembers;
