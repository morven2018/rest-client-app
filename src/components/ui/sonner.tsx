'use client';
import { useLocale } from 'next-intl';
import { useTheme } from 'next-themes';
import { Toaster as Sonner, ToasterProps, toast } from 'sonner';
import { Badge } from './badge';
import { dateString } from '@/lib/formatter';

const SHOW_DURATION = 2000;

const Toaster = ({ ...props }: ToasterProps) => {
  const { theme = 'system' } = useTheme();

  return (
    <Sonner
      theme={theme as ToasterProps['theme']}
      className="toaster group"
      style={
        {
          '--normal-bg': 'var(--popover)',
          '--normal-text': 'var(--popover-foreground)',
          '--normal-border': 'var(--border)',
        } as React.CSSProperties
      }
      {...props}
    />
  );
};

export interface ToastOptions {
  type?: 'success' | 'error' | 'note';
  additionalMessage?: string;
  action?: {
    label: string;
    onClick: () => void;
  };
  duration?: number;
  locale?: string;
}

export const useToast = () => {
  const locale = useLocale();

  const showToastWithLocale = (
    message: string,
    options?: Omit<ToastOptions, 'locale'>
  ) => {
    return showToast(message, { ...options, locale });
  };

  const toastSuccess = (
    message: string,
    options?: Omit<ToastOptions, 'type'>
  ) => showToastWithLocale(message, { ...options, type: 'success' });

  const toastError = (message: string, options?: Omit<ToastOptions, 'type'>) =>
    showToastWithLocale(message, { ...options, type: 'error' });

  const toastNote = (message: string, options?: Omit<ToastOptions, 'type'>) =>
    showToastWithLocale(message, { ...options, type: 'note' });

  return {
    showToast: showToastWithLocale,
    toastSuccess,
    toastError,
    toastNote,
  };
};

export const showToast = (message: string, options?: ToastOptions) => {
  const {
    type = 'note',
    additionalMessage,
    action,
    duration,
    locale = 'en',
  } = options || {};

  const getToastStyles = () => {
    switch (type) {
      case 'success':
        return {
          className: 'border-teal-100 shadow-inner bg-opacity-50',
          actionClass: 'bg-teal-700 hover:bg-teal-600 text-white',
        };
      case 'error':
        return {
          className: 'border-red-300 text-red-900 shadow-inner bg-opacity-50',
          actionClass: 'bg-red-600 hover:bg-red-700 text-white',
        };
      case 'note':
      default:
        return {
          className: 'border-blue-200 text-blue-900 shadow-inner bg-opacity-50',
          actionClass: 'bg-blue-600 hover:bg-blue-700 text-white',
        };
    }
  };

  const styles = getToastStyles();

  return toast.custom(
    (t) => (
      <div
        className={`p-5 rounded-lg border-2 ${styles.className} shadow-lg min-w-2xs max-w-lg flex flex-row gap-6 items-center m-0`}
      >
        <div className="flex flex-col gap-1">
          <div className="font-medium text-sm">{message}</div>

          {additionalMessage && (
            <div className="text-sm text-muted-foreground">
              <Badge variant={type === 'error' ? 'error' : 'ok'}>
                {additionalMessage}
              </Badge>
            </div>
          )}
          <div className="text-sm text-muted-foreground m-0">
            {dateString(new Date(), locale)}
          </div>
        </div>

        {action && (
          <button
            onClick={() => {
              action.onClick();
              toast.dismiss(t);
            }}
            className={`px-4 py-2 rounded-md text-xs font-medium transition-colors cursor-pointer ${styles.actionClass} min-w-[100px]`}
          >
            {action.label}
          </button>
        )}
      </div>
    ),
    {
      duration: duration || SHOW_DURATION,
    }
  );
};

export const toastSuccess = (
  message: string,
  options?: Omit<ToastOptions, 'type'>
) => showToast(message, { ...options, type: 'success' });

export const toastError = (
  message: string,
  options?: Omit<ToastOptions, 'type'>
) => showToast(message, { ...options, type: 'error' });

export const toastNote = (
  message: string,
  options?: Omit<ToastOptions, 'type'>
) => showToast(message, { ...options, type: 'note' });

export { Toaster, toast };
