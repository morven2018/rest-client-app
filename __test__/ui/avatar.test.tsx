import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { Avatar, AvatarImage, AvatarFallback } from '@/components/ui/avatar';

jest.mock('@/lib/utils', () => ({
  cn: (...classes: (string | undefined)[]) => classes.filter(Boolean).join(' '),
}));

// Mock @radix-ui/react-avatar
jest.mock('@radix-ui/react-avatar', () => {
  const originalModule = jest.requireActual('@radix-ui/react-avatar');
  return {
    ...originalModule,
    Root: ({
      className,
      children,
      'data-slot': dataSlot,
      ...props
    }: {
      className?: string;
      children: React.ReactNode;
      'data-slot'?: string;
    }) => (
      <div
        className={className}
        data-slot={dataSlot}
        data-testid="avatar-root"
        {...props}
      >
        {children}
      </div>
    ),
    Image: ({
      className,
      'data-slot': dataSlot,
      ...props
    }: {
      className?: string;
      'data-slot'?: string;
      src?: string;
      alt?: string;
    }) => (
      <img
        alt={props.alt}
        className={className}
        data-slot={dataSlot}
        data-testid="avatar-image"
        {...props}
      />
    ),
    Fallback: ({
      className,
      children,
      'data-slot': dataSlot,
      ...props
    }: {
      className?: string;
      children: React.ReactNode;
      'data-slot'?: string;
    }) => (
      <div
        className={className}
        data-slot={dataSlot}
        data-testid="avatar-fallback"
        {...props}
      >
        {children}
      </div>
    ),
  };
});

describe('Avatar Components', () => {
  describe('Avatar', () => {
    it('should render without throwing errors', () => {
      expect(() => render(<Avatar />)).not.toThrow();
    });

    it('should apply default className', () => {
      const { getByTestId } = render(<Avatar />);
      const avatarRoot = getByTestId('avatar-root');

      expect(avatarRoot.className).toContain('relative');
      expect(avatarRoot.className).toContain('flex');
      expect(avatarRoot.className).toContain('size-8');
      expect(avatarRoot.className).toContain('shrink-0');
      expect(avatarRoot.className).toContain('overflow-hidden');
      expect(avatarRoot.className).toContain('rounded-full');
    });

    it('should merge custom className with default classes', () => {
      const { getByTestId } = render(<Avatar className="custom-class" />);
      const avatarRoot = getByTestId('avatar-root');

      expect(avatarRoot.className).toContain('relative');
      expect(avatarRoot.className).toContain('custom-class');
    });

    it('should have correct data-slot attribute', () => {
      const { getByTestId } = render(<Avatar />);
      const avatarRoot = getByTestId('avatar-root');

      expect(avatarRoot.getAttribute('data-slot')).toBe('avatar');
    });

    it('should pass through additional props', () => {
      const { getByTestId } = render(
        <Avatar id="test-avatar" data-test="avatar-test" />
      );
      const avatarRoot = getByTestId('avatar-root');

      expect(avatarRoot.id).toBe('test-avatar');
      expect(avatarRoot.getAttribute('data-test')).toBe('avatar-test');
    });

    it('should render children', () => {
      const { getByText } = render(<Avatar>Test Content</Avatar>);
      expect(getByText('Test Content')).toBeTruthy();
    });
  });

  describe('AvatarImage', () => {
    it('should render without throwing errors', () => {
      expect(() =>
        render(<AvatarImage src="/test.jpg" alt="Test" />)
      ).not.toThrow();
    });

    it('should apply default className', () => {
      const { getByTestId } = render(
        <AvatarImage src="/test.jpg" alt="Test" />
      );
      const avatarImage = getByTestId('avatar-image');

      expect(avatarImage.className).toContain('aspect-square');
      expect(avatarImage.className).toContain('size-full');
    });

    it('should merge custom className with default classes', () => {
      const { getByTestId } = render(
        <AvatarImage src="/test.jpg" alt="Test" className="custom-class" />
      );
      const avatarImage = getByTestId('avatar-image');

      expect(avatarImage.className).toContain('aspect-square');
      expect(avatarImage.className).toContain('custom-class');
    });

    it('should have correct data-slot attribute', () => {
      const { getByTestId } = render(
        <AvatarImage src="/test.jpg" alt="Test" />
      );
      const avatarImage = getByTestId('avatar-image');

      expect(avatarImage.getAttribute('data-slot')).toBe('avatar-image');
    });

    it('should pass through image props', () => {
      const { getByTestId } = render(
        <AvatarImage
          src="/test.jpg"
          alt="Test Avatar"
          width={100}
          height={100}
        />
      );
      const avatarImage = getByTestId('avatar-image');

      expect(avatarImage.getAttribute('src')).toBe('/test.jpg');
      expect(avatarImage.getAttribute('alt')).toBe('Test Avatar');
    });
  });

  describe('AvatarFallback', () => {
    it('should render without throwing errors', () => {
      expect(() => render(<AvatarFallback>FB</AvatarFallback>)).not.toThrow();
    });

    it('should apply default className', () => {
      const { getByTestId } = render(<AvatarFallback>FB</AvatarFallback>);
      const avatarFallback = getByTestId('avatar-fallback');

      expect(avatarFallback.className).toContain('bg-muted');
      expect(avatarFallback.className).toContain('flex');
      expect(avatarFallback.className).toContain('size-full');
      expect(avatarFallback.className).toContain('items-center');
      expect(avatarFallback.className).toContain('justify-center');
      expect(avatarFallback.className).toContain('rounded-full');
    });

    it('should merge custom className with default classes', () => {
      const { getByTestId } = render(
        <AvatarFallback className="custom-class">FB</AvatarFallback>
      );
      const avatarFallback = getByTestId('avatar-fallback');

      expect(avatarFallback.className).toContain('bg-muted');
      expect(avatarFallback.className).toContain('custom-class');
    });

    it('should have correct data-slot attribute', () => {
      const { getByTestId } = render(<AvatarFallback>FB</AvatarFallback>);
      const avatarFallback = getByTestId('avatar-fallback');

      expect(avatarFallback.getAttribute('data-slot')).toBe('avatar-fallback');
    });

    it('should render children content', () => {
      const { getByText } = render(
        <AvatarFallback>Fallback Text</AvatarFallback>
      );
      expect(getByText('Fallback Text')).toBeTruthy();
    });

    it('should pass through additional props', () => {
      const { getByTestId } = render(
        <AvatarFallback id="test-fallback" data-test="fallback-test">
          FB
        </AvatarFallback>
      );
      const avatarFallback = getByTestId('avatar-fallback');

      expect(avatarFallback.id).toBe('test-fallback');
      expect(avatarFallback.getAttribute('data-test')).toBe('fallback-test');
    });
  });

  describe('Integration: Complete Avatar with Image and Fallback', () => {
    it('should render a complete avatar with image and fallback', () => {
      render(
        <Avatar>
          <AvatarImage src="/user.jpg" alt="User Avatar" />
          <AvatarFallback>UA</AvatarFallback>
        </Avatar>
      );

      const avatarRoot = screen.getByTestId('avatar-root');
      const avatarImage = screen.getByTestId('avatar-image');
      const avatarFallback = screen.getByTestId('avatar-fallback');

      expect(avatarRoot).toBeTruthy();
      expect(avatarImage).toBeTruthy();
      expect(avatarFallback).toBeTruthy();
      expect(avatarImage.getAttribute('src')).toBe('/user.jpg');
      expect(avatarImage.getAttribute('alt')).toBe('User Avatar');
      expect(avatarFallback.textContent).toBe('UA');
    });

    it('should apply custom classNames to all components', () => {
      const { getByTestId } = render(
        <Avatar className="avatar-custom">
          <AvatarImage src="/user.jpg" alt="User" className="image-custom" />
          <AvatarFallback className="fallback-custom">FB</AvatarFallback>
        </Avatar>
      );

      const avatarRoot = getByTestId('avatar-root');
      const avatarImage = getByTestId('avatar-image');
      const avatarFallback = getByTestId('avatar-fallback');

      expect(avatarRoot.className).toContain('avatar-custom');
      expect(avatarImage.className).toContain('image-custom');
      expect(avatarFallback.className).toContain('fallback-custom');
    });
  });

  describe('Edge Cases', () => {
    it('should handle undefined className gracefully', () => {
      const { getByTestId } = render(<Avatar className={undefined} />);
      const avatarRoot = getByTestId('avatar-root');

      expect(avatarRoot.className).toContain('relative');
    });

    it('should handle empty string className', () => {
      const { getByTestId } = render(<Avatar className="" />);
      const avatarRoot = getByTestId('avatar-root');

      expect(avatarRoot.className).toContain('relative');
    });
  });
});

describe('Avatar Components Snapshot', () => {
  it('Avatar should match snapshot', () => {
    const { container } = render(<Avatar />);
    expect(container).toMatchSnapshot();
  });

  it('AvatarImage should match snapshot', () => {
    const { container } = render(<AvatarImage src="/test.jpg" alt="Test" />);
    expect(container).toMatchSnapshot();
  });

  it('AvatarFallback should match snapshot', () => {
    const { container } = render(<AvatarFallback>FB</AvatarFallback>);
    expect(container).toMatchSnapshot();
  });

  it('Complete Avatar should match snapshot', () => {
    const { container } = render(
      <Avatar>
        <AvatarImage src="/user.jpg" alt="User" />
        <AvatarFallback>UA</AvatarFallback>
      </Avatar>
    );
    expect(container).toMatchSnapshot();
  });
});
