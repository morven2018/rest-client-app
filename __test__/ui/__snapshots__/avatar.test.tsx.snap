// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Avatar Components Snapshot Avatar should match snapshot 1`] = `
<div>
  <div
    class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
    data-slot="avatar"
    data-testid="avatar-root"
  />
</div>
`;

exports[`Avatar Components Snapshot AvatarFallback should match snapshot 1`] = `
<div>
  <div
    class="bg-muted flex size-full items-center justify-center rounded-full"
    data-slot="avatar-fallback"
    data-testid="avatar-fallback"
  >
    FB
  </div>
</div>
`;

exports[`Avatar Components Snapshot AvatarImage should match snapshot 1`] = `
<div>
  <img
    alt="Test"
    class="aspect-square size-full"
    data-slot="avatar-image"
    data-testid="avatar-image"
    src="/test.jpg"
  />
</div>
`;

exports[`Avatar Components Snapshot Complete Avatar should match snapshot 1`] = `
<div>
  <div
    class="relative flex size-8 shrink-0 overflow-hidden rounded-full"
    data-slot="avatar"
    data-testid="avatar-root"
  >
    <img
      alt="User"
      class="aspect-square size-full"
      data-slot="avatar-image"
      data-testid="avatar-image"
      src="/user.jpg"
    />
    <div
      class="bg-muted flex size-full items-center justify-center rounded-full"
      data-slot="avatar-fallback"
      data-testid="avatar-fallback"
    >
      UA
    </div>
  </div>
</div>
`;
